examples/cis2-multi-royalties/src/lib.rs

```diff
@@ -0,0 +1,1481 @@
+            owner: params.owner,
+        }))?;
+
+        // Metadata URL for the token.
+        logger.log(&Cis2Event::TokenMetadata::<_, ContractTokenAmount>(TokenMetadataEvent {
+            token_id,
+            metadata_url: MetadataUrl {
+                url:  build_token_metadata_url(&token_id),
+                hash: None,
+            },
+        }))?;
+    }
+    Ok(())
+}
+
+type TransferParameter = TransferParams<ContractTokenId, ContractTokenAmount>;
+
+/// Execute a list of token transfers, in the order of the list.
+///
+/// Logs a `Transfer` event and invokes a receive hook function for every
+/// transfer in the list.
+///
+/// It rejects if:
+/// - It fails to parse the parameter.
+/// - Any of the transfers fail to be executed, which could be if:
+///     - The `token_id` does not exist.
+///     - The sender is not the owner of the token, or an operator for this
+///       specific `token_id` and `from` address.
+///     - The token is not owned by the `from`.
+/// - Fails to log event.
+/// - Any of the receive hook function calls rejects.
+#[receive(
+    contract = "cis2_multi_royalties",
+    name = "transfer",
+    parameter = "TransferParameter",
+    error = "ContractError",
+    enable_logger,
+    mutable
+)]
+fn contract_transfer<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<State<S>, StateApiType = S>,
+    logger: &mut impl HasLogger,
+) -> ContractResult<()> {
+    // Parse the parameter.
+    let TransferParams(transfers): TransferParameter = ctx.parameter_cursor().get()?;
+    // Get the sender who invoked this contract function.
+    let sender = ctx.sender();
+
+    for Transfer {
+        token_id,
+        amount,
+        from,
+        to,
+        data,
+    } in transfers
+    {
+        let (state, builder) = host.state_and_builder();
+        // Authenticate the sender for this transfer
+        ensure!(from == sender || state.is_operator(&sender, &from), ContractError::Unauthorized);
+        let to_address = to.address();
+        let mut royalties: Option<CheckRoyaltyResult> = None;
+
+        if state.pay_royalty {
+            let royalty_parameters = CheckRoyaltyParams {
+                id:         token_id,
+                sale_price: amount.into(),
+            };
+            let royal_result = state.calculate_royalty_payments(royalty_parameters);
+            ensure!(royal_result.is_ok(), ContractError::InvalidTokenId);
+            royalties = Some(royal_result.unwrap());
+        }
+
+        // Update the contract state
+        state.transfer(&token_id, amount, &from, &to_address, builder, &royalties)?;
+
+        let mut receivers_amount = amount;
+        if royalties.is_some() {
+            // Log transfer event
+            logger.log(&Cis2Event::Transfer(TransferEvent {
+                token_id,
+                amount: concordium_cis2::TokenAmountU64(royalties.as_ref().unwrap().payment),
+                from: to_address,
```

Comment:
```suggestion
                from,
```
This should be the `from` address.

Reply:
Good spot